File: baekjoon_java/_10989.java
/*

10989번 - 수 정렬하기 3
https://www.acmicpc.net/problem/10989
분류 : 정렬

모든 수를 배열에 저장하고 정렬한 뒤 출력했다.
값이 10000이하이기 때문에 1~10000이 나온 횟수를 저장하는 배열을 만들고, 각 횟수만큼 인덱스를 출력하는 것이 더 빠르다.

*/

package baekjoon_java;

import java.io.*;
import java.util.Arrays;

public class _10989 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(br.readLine());
        }
        for (int num : Arrays.stream(nums).sorted().toArray()) {
            bw.write(num + "\n");
        }
        bw.flush();
    }
}

File: baekjoon_kotlin/_10000/10989.kt
package baekjoon_kotlin._10000

/*

10989번 - 수 정렬하기 3
https://www.acmicpc.net/problem/10989

모든 숫자를 입력받으며 크기가 10001인 배열에 각 숫자가 나온 횟수를 저장한다.
그리고 배열을 탐색하며 각 숫자가 나온 횟수만큼 그 숫자를 출력한다.
"내용".repeat(0)은 내용이 없기 때문에 나오지 않은 숫자는 출력되지 않는다.

*/

fun main() = System.`in`.bufferedReader().run {
    val count = IntArray(10001)
    val bw = System.out.bufferedWriter()
    repeat(readLine().toInt()) { count[readLine().toInt()]++ }
    for (i in 1..10000) bw.write("$i\n".repeat(count[i]))
    bw.flush()
}

/*

*/